### PR TITLE
Data cleanup enhancement

### PR DESCRIPTION
**Pull request by xaviermartinez2012**
Enhanced script by removing max run/clone arguments and using logic similar to python's `os.walk`.

---
This change is [Reviewable](https://reviewable.io/reviews/sorinlab/fah-data-cleanup/2)

### CODE REVIEW

File: fah-data-cleanup.pl [hunk 1]
```diff
@@ -5,44 +5,55 @@
...
-        `mv frame0.tpr temp`;
-        `rm *.tpr 2> /dev/null`;
-        `mv temp frame0.tpr`;
-        `mv ener.edr temp`;
-        `rm *.edr 2> /dev/null`;
-        `mv temp ener.edr`;
-        $currentclone++;
+my $projpath = $ARGV[0] or die HelpMessage();
+
+if (-d $projpath) {
```
Comment (khaister) [inline fah-data-cleanup.pl:10]: Invert `if` to reduce nesting.

File: fah-data-cleanup.pl [hunk 2]
```diff
@@ -5,44 +5,55 @@
...
+} else {
+    print STDOUT "$projpath: No such file or directory\n";
+    exit 1;
 }
-
 print STDOUT "Done!\n";
 
+# Arguments: pattern to search for, absolute path of directory
+sub pattern_walk {
+    my ($pattern, $path, @dirs, $num_dirs);
```
Comment (khaister) [inline fah-data-cleanup.pl:34]: Use first line in sub body to get parameters: `my ($pattern, $path) = @_;`. This, as a result, tells you your sub's parameters.

File: fah-data-cleanup.pl [hunk 3]
```diff
@@ -5,44 +5,55 @@
...
     }
-    $currentrun++;
+} else {
+    print STDOUT "$projpath: No such file or directory\n";
+    exit 1;
 }
-
 print STDOUT "Done!\n";
 
+# Arguments: pattern to search for, absolute path of directory
```
Comment (khaister) [inline fah-data-cleanup.pl:32]: Please remove this comment.

File: fah-data-cleanup.pl [hunk 4]
```diff
@@ -5,44 +5,55 @@
...
+sub pattern_walk {
+    my ($pattern, $path, @dirs, $num_dirs);
+    $pattern  = $_[0];
+    $path     = $_[1];
+    @dirs     = `ls $path | grep $pattern`;
```
Comment (khaister) [inline fah-data-cleanup.pl:37]: Please use `opendir()` and `readdir()`.

File: fah-data-cleanup.pl [hunk 5]
```diff
@@ -5,44 +5,55 @@
...
+    $num_dirs = scalar(@dirs);
+    for (my $x = 0 ; $x < $num_dirs ; $x++) {
+        my $dir = $dirs[$x];
+        chomp $dir;
```
Comment (khaister) [inline fah-data-cleanup.pl:41]: If you use `readdir()` there's no need to `chomp()`.

File: fah-data-cleanup.pl [hunk 6]
```diff
@@ -5,44 +5,55 @@
...
+}
+
 =head1 NAME
 
 fah-data-clean-up.pl - Remove unwanted files from a F@H dataset
 
 =head1 SYNOPSIS
 
-./fah-data-clean-up.pl <project_dir> <#_of_runs> <#_of_clones>
+./fah-data-clean-up.pl <Absolute path to PROJ# directory>
```
Comment (khaister) [inline fah-data-cleanup.pl:53]: Can I use relative path?

Also please use something like `<path_to_project_dir>` for style consistency in POD.

**Review by khaister — CHANGES_REQUESTED** [review]

File: fah-data-cleanup.pl (hunk 2 above)
Comment (khaister) [reply fah-data-cleanup.pl:34]: Please also rename the sub name to better show intention: this sub looks like it tries to **get dirs** that match a pattern. A language's characteristics (e.g. python's `os.walk()`) should not drive how we name things. We need to name things that's easy to understand and clearly shows meaning/intention.

File: fah-data-cleanup.pl [hunk 7]
```diff
@@ -5,44 +5,55 @@
...
+            `mv temp frame0.tpr`;
+            `mv ener.edr temp`;
+            `rm *.edr 2> /dev/null`;
+            `mv temp ener.edr`;
+        }
     }
-    $currentrun++;
+} else {
+    print STDOUT "$projpath: No such file or directory\n";
+    exit 1;
```
Comment (khaister) [inline fah-data-cleanup.pl:28]: Please use `die` instead of `print STDOUT` + `exit 1`.

**Comment by khaister** [comment]
Hi @xaviermartinez2012, any updates on this?

**Comment by xaviermartinez2012** [comment]
@khaister Lets close this pull request. The changes weren't really _that_ significant.

**Pull request closed without merging**